File: NET/Kata.RockPaperScissor/Rock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kata.RockPaperScissor
{
    public class Rock : Shape
    {
        public Rock()
        {
            ShapeType = ShapeType.Rock;
            Name = "Rock";
            winsTo = new List<ShapeType>() { ShapeType.Scissor, ShapeType.Lizard };
        } 
    }
}

File: JavaScript/src/objects/rock.js
var Shape = require('./shape');

var Rock = Object.create(Shape);

Rock.setup = function (ShapeType) {
	this.init({
		shapeType: ShapeType.rock,
		winsTo: [ShapeType.scissor, ShapeType.lizard]
	});
}

module.exports = Rock;